src/utils/api.js

```diff
@@ -31,6 +42,103 @@ export const formatActions = (actionsByDayObject) => {
   });
 };
 
+// Takes array of action objects and returns an object with {key: value} pairs of {hourOfDay: #-of-actions}
+export const getActionsByTimeOfDay = (actions) => {
+  const actionsByTimeOfDay = {
+    [LATE_NIGHT]: 0,
+    [EARLY_MORNING]: 0,
+    [MORNING]: 0,
+    [AFTERNOON]: 0,
+    [EVENING]: 0,
+    [NIGHT]: 0,
+  };
+  actions.forEach((action) => {
+    const actionHourOfDay = action.createdAt.slice(11, 13);
+    if (actionHourOfDay >= 0 && actionHourOfDay < 4) {
+      actionsByTimeOfDay[LATE_NIGHT] += 1;
+    } else if (actionHourOfDay >= 4 && actionHourOfDay < 8) {
+      actionsByTimeOfDay[EARLY_MORNING] += 1;
+    } else if (actionHourOfDay >= 8 && actionHourOfDay < 12) {
+      actionsByTimeOfDay[MORNING] += 1;
+    } else if (actionHourOfDay >= 12 && actionHourOfDay < 16) {
+      actionsByTimeOfDay[AFTERNOON] += 1;
+    } else if (actionHourOfDay >= 16 && actionHourOfDay < 20) {
+      actionsByTimeOfDay[EVENING] += 1;
+    } else if (actionHourOfDay >= 20 && actionHourOfDay < 24) {
+      actionsByTimeOfDay[NIGHT] += 1;
+    } else {
+      // eslint-disable-next-line no-console
+      console.log(`actionHourOfDay ${actionHourOfDay} is undefined`);
+    }
+  });
+  return actionsByTimeOfDay;
+};
+
+// Takes object with {key: value} pairs of {timeOfDay: #-of-actions}
+// returns a date-sorted array in Recharts.js format
+export const formatActionsByTimeOfDay = (actionsByTimeOfDayObject) => {
+  const actionsByTimeOfDayArray = Object.entries(actionsByTimeOfDayObject);
+  return actionsByTimeOfDayArray.map((entry) => {
+    return {
+      timeOfDay: entry[0],
+      count: entry[1],
+    };
+  });
+};
+
+// Takes array of action objects and returns an object with {key: value} pairs of {verb: %-of-actions}
+export const getActionsByVerb = (actions) => {
+  const totalActions = actions.length;
+  const actionsByVerb = {};
+  actions.forEach((action) => {
+    if (!actionsByVerb[action.verb]) {
+      // if verb is still not in the actionsByVerb object, add it and assign it to (1 / totalActions)
+      // we use (1 / totalActions) because in the end we want this object to be {verb: PERCENTAGE-of-total-actions}
+      actionsByVerb[action.verb] = 1 / totalActions;
+    } else {
+      actionsByVerb[action.verb] += 1 / totalActions;
```

Comment:
What is not clear here is the precedence of the = and the /. Wrap `(1 / ...)`.

Reply:
I actually did this but just noticed that prettier removes it. Any thoughts?